src/likelihoods/negativebinomial.jl

```diff
@@ -0,0 +1,25 @@
+"""
+    NegativeBinomialLikelihood(l=logistic; successes::Real=1)
+
+Negative binomial likelihood with number of successes `successes`.
```

Comment:
I come a bit late, but I disagree with the use of `successes` literally no other distributions has such explicit naming.

Reply:
Happy to revert, let me know when you reach a consensus.

Reply:
@devmotion what is the motivation behind the verbose argument? To make it clear it's different from Wikipedia?

Reply:
@devmotion what is the motivation behind the verbose argument? To make it clear it's different from Wikipedia?

Reply:
Mainly to make it clear what the parameter means, similar to how we renamed some keyword arguments in KernelFunctions - `r` is not very descriptive I'd say 😅 And, of course, there's an additional source of confusion here due to the discrepancy with Wikipedia that could maybe be resolved by a more descriptive name.

Reply:
Should we just do what we did for KF.jl, i.e. having 2 keyword arguments for the same thing as `... ; r, successes=r)` ?

Reply:
> `r` is not very descriptive,

Just as mu and sigma yet everyone uses that. I think it's a matter of being familiar with the negative binomial distribution.

Reply:
> > `r` is not very descriptive,
> 
> Just as mu and sigma yet everyone uses that. I think it's a matter of being familiar with the negative binomial distribution. 

I don't think anyone uses `mu` and `sigma` as parameters of a normal distribution for anything else than mean and standard deviation. But apparently there are different interpretations of `r` in a negative binomial distribution, as Wikipedia and Distributions show.

Reply:
How about going crazy and starting here :)

Reply:
While I don't particularly care either way, IMHO users will usually consult the documentation when playing around with different distributions (even just to check which distributions are actually supported), so they will likely find out what `r` means.

Reply:
Plus, for end users who are fitting a `LatentGP` with negative binomial likelihood, actually it does not matter that much what the parameters mean until they are plotting the results. In that case, one could define some pretty printing facilities that include a parameter table with some descriptors.

Reply:
> While I don't particularly care either way, IMHO users will usually consult the documentation when playing around with different likelihoods (even just to check which likelihoods are actually supported), so they will likely find out what r means.

IMO the main motivation is not to simplify coding since, as you say, usually one has to consult the documentation anyway. I think the main advantage of more descriptive arguments is the same as for e.g. descriptive function and variable names: it makes it easier to (re-)read the code and to reason about it.

Reply:
Alright, alright, let's stick with `successes`, at least it will force people using `r=` to check the docs and see the warning